win32ss/gdi/gdi32/misc/misc.c

```diff
@@ -745,7 +745,7 @@ GdiFixUpHandle(HGDIOBJ hGdiObj)
     Entry = GdiHandleTable + GDI_HANDLE_GET_INDEX(hGdiObj);
 
     /* Rebuild handle for Object */
-    return hGdiObj = (HGDIOBJ)(((LONG_PTR)(hGdiObj)) | (Entry->Type << GDI_ENTRY_UPPER_SHIFT));
+    return hGdiObj = (HGDIOBJ)(((ULONG_PTR)(hGdiObj)) | (Entry->Type << GDI_ENTRY_UPPER_SHIFT));
```

Comment:
I'm not sure I understand this change, or how it relates to the newly added tests.
Somehow `(Entry->Type << GDI_ENTRY_UPPER_SHIFT)` got sign-extended before? What are the consequences of this, especially given that your test seems to show that the upper 32 bits are ignored?